src/d3d9/d3d9_device.cpp

```diff
@@ -6908,7 +6908,7 @@ namespace dxvk {
       const uint32_t buffersToUpload = m_activeVertexBuffersToUpload & usedBuffersMask;
       for (uint32_t bufferIdx : bit::BitMask(buffersToUpload)) {
         auto* vbo = GetCommonBuffer(m_state.vertexBuffers[bufferIdx].vertexBuffer);
-        if (vbo != nullptr && vbo->NeedsUpload())
+        if (likely(vbo != nullptr && vbo->NeedsUpload()))
```

Comment:
why is this likely?

Reply:
Because we're inside the bit mask loop. Any 1 bit of that should also have NeedsUpload() set unless.